cmd/argocd/commands/app.go

```diff
@@ -949,7 +960,13 @@ func getLocalObjects(app *argoappv1.Application, local, appLabelKey, kubeVersion
 
 func getLocalObjectsString(app *argoappv1.Application, local, appLabelKey, kubeVersion string, kustomizeOptions *argoappv1.KustomizeOptions,
 	configManagementPlugins []*argoappv1.ConfigManagementPlugin) []string {
-	res, err := repository.GenerateManifests(local, "/", app.Spec.Source.TargetRevision, &repoapiclient.ManifestRequest{
+
+	// Get repository root using git
+	repoRootBytes, err := exec.Command("git", "rev-parse", "--show-toplevel").Output()
```

Comment:
Please do not use `os/exec` package, rather use `github.com/argoproj/pkg/utils/exec` as it is the package we use throughout ArgoCD and has some safe-guards built-in.

Also, I think the path to the repository root should be available from `github.com/argoproj/argo-cd/util/git.Root()` already.

Reply:
Ah, that was a quick shot. Haven't noticed that this is in the CLI code. I wonder whether we should make the requirements for local objects to be a Git repository.

Reply:
How about treating `--jsonnet-libs` relative to the current working directory when in `local` mode instead?

Reply:
I was also thinking on just using it relative to the current working dir but that creates the issue of an inconsistent workflow.
E.g. having a `shared` library folder at the repository root which is consumed by many different applications in subdirectories would require a change in `--jsonnet-libs` when local syncing.

Not sure what's the best approach. Maybe passing a global flag to the CLI with `--local-repo-root`? (Which then would require manual input instead of it being resolved via an exec)

Reply:
I'll address the switch to the `github.com/argoproj/pkg/utils/exec` package as soon as there's a decision!

Reply:
Hm, I think we should take care not to introduce a new mandatory switch or otherwise backwards-incompatible change (such as assuming that the local path would be a Git repo). This could break existing workflows for people using local sync in their CI or other scripts.

I think, if we introduce a new switch, it should only be mandatory if the local sources indeed require this, for example if the application to be locally synced has a non-empty value for jsonnet libs path specified. I also think the name of the parameter should make clear, that it's for jsonnet sources only, maybe `--jsonnet-local-root-path`. Unless it could be of value to other types of sources, too.

Reply:
Okay, good point you have there.
I'll go with `--jsonnet-local-root-path` - this is the clearest solution IMO

Reply:
Possibly found a nicer way without breaking backwards compatibility, see the new commit & feel to provide your opinion